python/cudf/cudf/core/column/string.py

```diff
@@ -5855,7 +5856,14 @@ def __init__(
     ):
         if not isinstance(data, Buffer):
             raise ValueError("data must be a Buffer")
-        if dtype != CUDF_STRING_DTYPE:
+        if (
+            dtype != CUDF_STRING_DTYPE
+            and not isinstance(dtype, pd.StringDtype)
+            and not (
+                isinstance(dtype, pd.ArrowDtype)
+                and dtype.pyarrow_dtype == pa.string()
```

Comment:
Can you also include `pa.large_string()`?

Reply:
Done.